### DISCUSSION TITLE
Unleash client v3.3.0 compatibility with Unleash server v5.0.0

### DISCUSSION THREAD

**Discussion by EmadBinAbid** [Unleash/unleash#3716] [General]
I've been using Unleash server v4.16 for almost a year, which is smooth and compatible with Unleash client v3.3.0. 

Now, I'm planning to upgrade the server to v5.0.0 while the client still stays at v3.3.0.

I wanted to ask if this flow is fine or do I have to upgrade the client as well? 

@andreas-unleash @ivarconr

**Comment by pransh15** [comment]
Hey @EmadBinAbid 👋 

That's a good question! Unleash follows semantic versioning and strives to maintain backward compatibility within major versions (e.g. 3.x.x to 3.y.y) so it usually is fine. Since you are upgrading to the newly introduced Unleash v5.0, there are some changes you need to know before you upgrade:

- Any version of node.js that does not include LTS. In other words, Unleash will only work with node.js version 18 and above.
- Auth with Google SSO. Reason: Google now supports OIDC out of the box. If this breaks for you, let us know.
- “password” as the default password for Unleash DB connections. It’s now a more appropriate–and English-friendly–“password”. Production shouldn’t be impacted. This should only affect dev build processes, as well as basic setup.
- Programs with APIs that depend on the shape of invalid data errors. We’ve changed our error message format, which means the [shape has changed.](https://docs.getunleash.io/reference/deploy/migration-guide)

More about breaking changes can be found in our [migration guide.](https://docs.getunleash.io/reference/deploy/migration-guide) 🔗

**Comment by ivarconr** [comment]
> I wanted to ask if this flow is fine or do I have to upgrade the client as well?

No need to upgrade the client, should be compatible with Unleash v5. The SDK APIs are fully compatible with and the same as for v4.

BTW: Would be good if you also include the SDK name (Java, Node, Ruby or similar)
